src/Gauge.php

```diff
@@ -0,0 +1,117 @@
+<?php declare(strict_types=1);
+
+namespace JTL\GoPrometrics\Client;
+
+use GuzzleHttp\ClientInterface;
+
+/**
+ * This File is part of JTL-Software
+ *
+ * User: Milanowicz
+ * Date: 2020-05-07
+ */
+class Gauge
+{
+    private ClientInterface $client;
+    private string $baseUrl;
+
+    public function __construct(ClientInterface $client, string $baseUrl)
+    {
+        $this->client = $client;
+        $this->baseUrl = $baseUrl;
+    }
+
+    /**
+     * @param string $namespace
+     * @param string $name
+     * @param LabelList|null $tagList
+     * @param string $help
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     */
+    public function inc(
+        string $namespace,
+        string $name,
+        LabelList $tagList = null,
+        string $help = ''
+    ): void {
+        $this->sendRequest($namespace, $name, 1, $tagList, $help);
+    }
+
+    /**
+     * @param string $namespace
+     * @param string $name
+     * @param float $value
+     * @param LabelList|null $tagList
+     * @param string $help
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     */
+    public function incBy(
+        string $namespace,
+        string $name,
+        float $value,
+        LabelList $tagList = null,
+        string $help = ''
+    ): void {
+        $this->sendRequest($namespace, $name, $value, $tagList, $help);
+    }
+
+    /**
+     * @param string $namespace
+     * @param string $name
+     * @param LabelList|null $tagList
+     * @param string $help
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     */
+    public function dec(
+        string $namespace,
+        string $name,
+        LabelList $tagList = null,
+        string $help = ''
+    ): void {
+        $this->sendRequest($namespace, $name, -1, $tagList, $help);
+    }
+
+    /**
+     * @param string $namespace
+     * @param string $name
+     * @param float $value
+     * @param LabelList|null $tagList
+     * @param string $help
+     * @throws \GuzzleHttp\Exception\GuzzleException
+     */
+    public function decBy(
+        string $namespace,
+        string $name,
+        float $value,
+        LabelList $tagList = null,
+        string $help = ''
+    ): void {
+        $this->sendRequest($namespace, $name, $value * (-1), $tagList, $help);
```

Comment:
`decBy("foo", "bar", -2)` ;)

What about implement only one method `add()` instead of implemting decBy and incBy?

Reply:
Okay good point and reason why is that:
https://github.com/prometheus/client_python
https://github.com/Superbalist/laravel-prometheus-exporter

Python and PHP has this Methods. I would prefer to keep it little bit equal to Python. I could change and or add method set or rename to inc for counter too. For Gauge I implements the method set too.

That's the reason why too I named at Histogram the method observe, because other Libs.

Reply:
Okay, lets go this way